packages/tools/viewer/src/viewer.ts

```diff
@@ -1411,9 +1608,127 @@ export class Viewer implements IDisposable {
+            // and the model may not even be in view. So when this is the case, by default we reframe the camera.
+            reframe = this.selectedAnimation !== (this._options?.selectedAnimation ?? 0);
+        }
+
+        if (reframe) {
+            this._reframeCamera(true);
+        } else {
+            this._reset(true, "camera");
+        }
+    }
+
+    /**
+     * Updates the camera pose.
+     * @param pose The new pose of the camera.
+     * @remarks Any unspecified values are left unchanged.
+     */
+    public updateCamera(pose: { alpha?: number; beta?: number; radius?: number; targetX?: number; targetY?: number; targetZ?: number }): void {
+        // undefined means default for _resetCameraFromBounds, so convert to NaN if needed.
+        this._reframeCameraFromBounds(
+            true,
+            this._loadedModels,
+            pose.alpha ?? NaN,
+            pose.beta ?? NaN,
+            pose.radius ?? NaN,
+            pose.targetX ?? NaN,
+            pose.targetY ?? NaN,
+            pose.targetZ ?? NaN
+        );
+    }
+
+    /**
+     * Resets the viewer to its initial state based on the options passed in to the constructor.
+     * @param flags The flags that specify which parts of the viewer to reset. If no flags are provided, all parts will be reset.
      */
-    public resetCamera() {
-        this._camera.restoreState();
+    public reset(...flags: ResetFlag[]) {
+        this._reset(true, ...flags);
+    }
+
+    private _reset(interpolate: boolean, ...flags: ResetFlag[]) {
+        const handleLoadError = async (promise: Promise<unknown>) => {
+            try {
+                await promise;
+            } catch (error) {
+                if (!(error instanceof AbortError)) {
+                    Logger.Error(error);
+                }
+            }
+        };
+
+        if (flags.length === 0 || flags.includes("source")) {
+            handleLoadError(this._updateModel(this._options?.source));
+        }
+
+        if (flags.length === 0 || flags.includes("environment")) {
+            this._scene.clearColor = new Color4(...(this._options?.clearColor ?? DefaultViewerOptions.clearColor));
+            this.environmentConfig = {
+                intensity: this._options?.environmentConfig?.intensity ?? DefaultViewerOptions.environmentConfig.intensity,
+                blur: this._options?.environmentConfig?.blur ?? DefaultViewerOptions.environmentConfig.blur,
+                rotation: this._options?.environmentConfig?.rotation ?? DefaultViewerOptions.environmentConfig.rotation,
+                visible: this._options?.environmentConfig?.visible ?? DefaultViewerOptions.environmentConfig.visible,
+            };
+            if (this._options?.environmentLighting === this._options?.environmentSkybox) {
+                handleLoadError(this._updateEnvironment(this._options?.environmentLighting, { lighting: true, skybox: true }, this._loadEnvironmentAbortController?.signal));

```

Comment:
Same comment as before. If any of these promises fail there is no error handling, and they will run after this function will be done (and "in parallel")

Reply:
Same response as above :)